lib/reek/configuration/app_configuration.rb

```diff
@@ -90,10 +90,11 @@ def self.new(*)
       #
       # @param source_via [String] - the source of the code inspected
       #
-      # @return [Hash] the directory directive for the source or, if there is
-      # none, the default directive
+      # @return [Hash] the directory directive for the source with the default directive
+      #                reverse-merged into it.
       def directive_for(source_via)
-        directory_directives.directive_for(source_via) || default_directive
+        hit = directory_directives.directive_for(source_via)
+        hit ? default_directive.merge(hit) : default_directive
```

Comment:
>> Can we make directory_directives.directive_for return an empty hash ({}) and then just merge that unconditionally?

I thought about this as well but it kind of hurts my feelings. 
From a type perspective this seems wrong, very wrong. Think about it: I ask "Give me a directive for this directory". 
If there is none, I'd expect something that signals absence. So, nil. An empty hash doesn't mean anything and would be confusing in a lot of context. We'd do this **only** for convenience and only here. But we would kill semantical sense for that.
WDYT?